backend/src/nodes/nodes/image_utility/convolve.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import cv2
+import numpy as np
+
+from . import category as ImageUtilityCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, NumberInput, NoteTextAreaInput
+from ...properties.outputs import ImageOutput
+from ...properties import expression
+
+
+@NodeFactory.register("chainner:image:image_convolve")
+class ImageConvolveNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Convolves input image with input kernel"
+        self.inputs = [
+            ImageInput("Grayscale Image", channels=1),
+            NoteTextAreaInput("Kernel String"),
+            NumberInput("Padding", minimum=0, default=0),
+            NumberInput("Strides", minimum=1, default=1),
+        ]
+        self.outputs = [
+            ImageOutput(
+                image_type="""
+                    let w = Input0.width;
+                    let h = Input0.height;
+
+                    let kernel = Input1;
+
+                    let padding = Input2;
+                    let stride = Input3;
+
+                    Image {
+                        width: int & floor((w + padding * 2 - 1) / stride + 1),
+                        height: int & floor((h + padding * 2 - 1) / stride + 1),
+                        channels: 1,
+                    }
+                """,
+                channels=1,
+            )
+        ]
+        self.category = ImageUtilityCategory
+        self.name = "Convolve"
+        self.icon = "MdAutoFixHigh"
+        self.sub = "Miscellaneous"
+
+    def run(
+        self,
+        img: np.ndarray,
+        kernel_in: str,
+        padding: int,
+        strides: int,
```

Comment:
Stride still has to be implemented. This can be done like this `output[::stride, ::stride, ...]` after the convolution.

That being said, do we really need stride? It's essentially just nearest neighbor downsampling. 
Also @joeyballentine

Reply:
I suppose we don't

Reply:
I just removed the strides so it should be fine now.